server/lib/watcher.js

```diff
@@ -0,0 +1,182 @@
+const logger = require("pino")();
+import BN from "bn.js";
+
+import Poller from "../lib/poller";
+import { leap } from "../middleware/leap";
+import { twilioClient } from "../middleware/twilio";
+
+import config from "../../src/config";
+import ERC20Abi from "../abis/ERC20";
+import redis from "redis";
+
+class Watcher {
+  constructor() {
+    this.state = {
+      uTxoSum: 0,
+      exitSum: 0,
+      plasmaBalance: 0
+    };
+
+    this.init = this.init.bind(this);
+    this.redis = redis.createClient();
+  }
+
+  async init() {
+    const poller = new Poller();
+
+    poller.onPoll(async () => {
+      logger.info("##### STARTING PLASMA CHECK #####");
+      logger.info("");
+      logger.info(await this.validateLeapPlasma());
+      logger.info("##### FINISHED PLASMA CHECK #####");
+      poller.poll(config.pollingDelay);
+    });
+
+    if (leap.web3) {
+      poller.poll();
+    } else {
+      setTimeout(this.init, 500);
+    }
+  }
+
+  async getUnspentTransactionSum() {
+    const { bridge, exitHandler, plasma, web3 } = leap;
+    const exits = [];
+
+    const uTxos = await plasma.getUnspentAll();
+    logger.info(`UNSPENT TRANSACTIONS: ${uTxos.length}`);
+
+    const addresses = await plasma.getColors();
+    logger.info(`ADDRESSES: ${addresses.length}`);
+
+    const colors = await Promise.all(
+      addresses.map(async address => {
+        return plasma.getColor(address);
+      })
+    );
+    logger.info(`COLORS: ${colors.length}`);
+
+    const uTxoSums = await Promise.all(
+      colors.map(async color => {
```

Comment:
this will do `colors.length * utxos.length` number of iterations, while it is enough to do just `utxos.length` iterations to do the same. E.g. as a `utxos.reduce` reducing to a map `{ color: balance }`